### DISCUSSION TITLE
VS 2022 The type or namespace name 'Windows' could not be found

### DISCUSSION THREAD

**Discussion by ryankeithster** [microsoft/CsWin32#885] [Q&A] [answered]
I saw the previous discussion that was having this issue when just trying to get started and as far as I can tell I'm matching the recommended solution, but I'm still getting the error "The type or namespace name 'Windows' could not be found".
I'm using Visual Studio 2022. and I've added the CsWin32 Nuget package to my C# console app.

```
using Windows.Win32;

PInvoke.CreateFile(/*args*/);
```

**Comment by AArnott** [comment, marked as answer]
Have you created a NativeMethods.txt file with `CreateFile` in it?

**Comment by ryankeithster** [comment]
Wow that fixed it. That's pretty interesting to me how the presence or lack of this text file can affect the compilation. I guess I have some learning to do about the source generation workflow. Thank you for answering my dumb question!

**Reply by AArnott** [reply]
tip: next time post your reply as a comment to the answer instead of as its own answer. :)

Without that file, CsWin32 has nothing to generate. That's why the namespace doesn't exist.
